browser/src/Services/Commands/GlobalCommands.ts

```diff
@@ -45,6 +50,78 @@ export const activate = (
     const popupMenuPrevious = popupMenuCommand(() => menuManager.previousMenuItem())
     const popupMenuSelect = popupMenuCommand(() => menuManager.selectMenuItem())
 
+    const gotoNextError = async () => {
+        const errors = getDiagnosticsInstance().getErrors()
+        const activeBuffer = editorManager.activeEditor.activeBuffer
+        const currentFileErrors = getAllErrorsForFile(activeBuffer.filePath, errors)
+        const currentPosition = activeBuffer.cursor
+
+        if (!currentFileErrors || currentFileErrors.length === 0) {
+            return
+        }
+
+        for (const error of currentFileErrors) {
+            if (isInRange(currentPosition.line, currentPosition.column, error.range)) {
+                continue
+            }
+
+            const currentLine = (await activeBuffer.getLines(currentPosition.line))[0]
+            if (
+                currentPosition.line === error.range.start.line &&
+                currentLine.length <= error.range.start.character
+            ) {
+                continue
+            }
+
+            if (
+                error.range.start.line > currentPosition.line ||
+                (error.range.start.line === currentPosition.line &&
+                    error.range.start.character > currentPosition.column)
+            ) {
+                await activeBuffer.setCursorPosition(
+                    error.range.start.line,
+                    error.range.start.character,
+                )
+                return
+            }
+        }
+
+        activeBuffer.setCursorPosition(
+            currentFileErrors[0].range.start.line,
+            currentFileErrors[0].range.start.character,
+        )
+    }
+
+    const gotoPreviousError = async () => {
+        const errors = getDiagnosticsInstance().getErrors()
+        const activeBuffer = editorManager.activeEditor.activeBuffer
+        const currentFileErrors = getAllErrorsForFile(activeBuffer.filePath, errors)
+        const currentPosition = activeBuffer.cursor
+
+        if (!currentFileErrors || currentFileErrors.length === 0) {
+            return
+        }
+
+        let lastError = currentFileErrors[currentFileErrors.length - 1]
+        for (const error of currentFileErrors) {
+            if (
+                isInRange(currentPosition.line, currentPosition.column, error.range) ||
+                error.range.start.line > currentPosition.line ||
+                (error.range.start.line === currentPosition.line &&
+                    error.range.start.character > currentPosition.column)
+            ) {
+                await activeBuffer.setCursorPosition(
+                    lastError.range.start.line,
+                    lastError.range.start.character,
+                )
+                return
+            }
+            lastError = error
+        }
+
+        activeBuffer.setCursorPosition(lastError.range.start.line, lastError.range.start.character)
```

Comment:
not sure if this is async but its likely to be in which case does it need to be awaited since its not being returned to be awaited elsewhere